samza-sql/src/main/java/org/apache/samza/sql/fn/GetSqlFieldUdf.java

```diff
@@ -51,22 +52,21 @@
  *           - sessionKey (Scalar)
  *
  */
-@SamzaSqlUdf(name = "GetSqlField")
+@SamzaSqlUdf(name = "GetSqlField", description = "Get an element from complex Sql field.")
```

Comment:
GetSqlField is kind of limiting as it returns only a String. For instance, we cannot get a sub-record or an array or any other type. Just wondering if other cases work if we return an object from execute ? Or since now we have ability to add more methods to an Udf (although I'm still not clear on how these different methods can be invoked in Udf unless they are fn overloads), is there any other way we could make it return different types ?

Reply:
Function overloads will not work if only the return types change. They work if the arguments are different. If a UDF has to return a type that can be different, Then it has to return an object in which case the return type will be "ANY". Does it answer your question?

Reply:
Sure. Could we change the description to reflect that it returns a string. May be: "Get an element as a String from Complex Sql field." ? As a separate effort, we will have to add another such Udf that returns ANY.